wfl/Models/StatusSearchCriteria.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+using MongoDB.Driver;
+using System.Threading.Tasks;
+
+namespace Symphono.Wfl.Models
+{
+    public class StatusSearchCriteria: ICriteria<FoodOrder>
+    {
+        public FoodOrder.StatusOptions Status { get; set;}
+
+        public async Task<IEnumerable<FoodOrder>> ApplyCriteria(IMongoCollection<FoodOrder> collection)
+        {
+            IAsyncCursor<FoodOrder> task = await collection.FindAsync(o => o.Status == this.Status);
+            IList<FoodOrder> orders = task.ToList();
```

Comment:
This is likely an evolution. I asked Tej to look and see if the mongo library we exposed allows LINQ to Mongo and then we could pass in and return an `IQueryable<TEntity>`. Personally, I am ok with some copy paste for the time being until we understand more of how criteria can be applied to mongo queries. We also should understand if these are DELAYED execution queries with the `IAsyncCursor<T>` or it pulls all the data and queries in memory -- hopefully NOT the later.